### DISCUSSION TITLE
Add a way to exclude pages from the sitemap

### DISCUSSION THREAD

**Discussion by edazpotato** [withastro/roadmap#53] [Proposal]
This is useful for things like redirect pages that we don't really want to be encouraging search engines to index.
There's a TODO commment for this [here](https://github.com/withastro/astro/blob/cc1dae55c8bbf0a7d862e227f7daed138c485be4/packages/astro/src/core/ssr/sitemap.ts#L5).

**Comment by dguo** [comment]
I would also find this useful for being able to publish a page that is technically public but doesn't have any links to it and ideally wouldn't show up in search results. For example, I'd like to be able to publish a draft blog post for someone to easily review, but I don't want the post to be broadly accessible until I'm ready.

Right now, I could accomplish that by [filtering](https://docs.astro.build/en/guides/integrations-guide/sitemap/#filter) on the page's URL, but that feels pretty clunky. If the filter functionality allowed me to access Markdown frontmatter, then I could set a `draft` property for the sitemap integration to read.

**Comment by dguo** [comment]
Hi @florian-lefebvre, I saw that you closed this as resolved. Was that because this feature is available now? I skimmed the changelog but didn't see anything related to this request.

**Reply by florian-lefebvre** [reply]
Hi sorry, I was cleaning up the oldest proposals, I'm happy to reopen! FYI you can achieve this using https://inox-tools.fryuni.dev/sitemap-ext
